### ISSUE TITLE
rmlmapper with spring boot

### ISSUE THREAD

**Issue by istoony** [RMLio/rmlmapper-java#42]
Hi everyone,
I'm trying to integrate the library in a spring boot application. The problem is that there always are conflicts with dependency that I can not solve. In particular the stack trace I have is:

![Schermata del 2019-10-15 21-38-40](https://user-images.githubusercontent.com/16716284/66863850-2f57eb00-ef94-11e9-92bf-34a0fe448e40.png)

POM:
	
```
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>2.1.3.RELEASE</version>
		<relativePath /> <!-- lookup parent from repository -->
	</parent>

	<groupId>cefriel</groupId>
	<artifactId>semanticfuelserver</artifactId>
	<version>0.0.1-SNAPSHOT</version>
	<packaging>jar</packaging>
	<url>http://maven.apache.org</url>

	<properties>
		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
		<java.version>8</java.version>
		<jackson.version>2.9.8</jackson.version>
		<junit.jupiter.version>5.5.2</junit.jupiter.version>
	</properties>

	<repositories>
		<repository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>https://repo.spring.io/milestone</url>
		</repository>
		<repository>
			<id>jitpack.io2</id>
			<url>https://jitpack.io</url>
		</repository>
	</repositories>
	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-thymeleaf</artifactId>
			<exclusions>
				<exclusion>
					<groupId>org.slf4j</groupId>
					<artifactId>slf4j-api</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter</artifactId>
			<exclusions>
				<exclusion>
					<groupId>org.springframework.boot</groupId>
					<artifactId>spring-boot-starter-logging</artifactId>
				</exclusion>

			</exclusions>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-autoconfigure</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-webflux</artifactId>
			<exclusions>
				<exclusion>
					<groupId>org.hibernate.validator</groupId>
					<artifactId>hibernate-validator</artifactId>
				</exclusion>
				<exclusion>
					<groupId>org.slf4j</groupId>
					<artifactId>slf4j-api</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-log4j2</artifactId>
			<exclusions>
				<exclusion>
					<groupId>org.slf4j</groupId>
					<artifactId>slf4j-api</artifactId>
				</exclusion>
			</exclusions>
		</dependency>

		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-devtools</artifactId>
			<optional>true</optional>
		</dependency>
		<dependency>
			<groupId>com.fasterxml.jackson.core</groupId>
			<artifactId>jackson-databind</artifactId>
		</dependency>
		<dependency>
			<groupId>com.fasterxml.jackson.dataformat</groupId>
			<artifactId>jackson-dataformat-yaml</artifactId>
		</dependency>
		<dependency>
			<groupId>org.webjars</groupId>
			<artifactId>bootstrap</artifactId>
			<version>3.3.7</version>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-test</artifactId>
			<scope>test</scope>
			<exclusions>
				<exclusion>
					<groupId>junit</groupId>
					<artifactId>junit</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		<dependency>
			<groupId>org.junit.platform</groupId>
			<artifactId>junit-platform-launcher</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.junit.jupiter</groupId>
			<artifactId>junit-jupiter-api</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>io.projectreactor</groupId>
			<artifactId>reactor-test</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.springframework.security</groupId>
			<artifactId>spring-security-test</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.junit.jupiter</groupId>
			<artifactId>junit-jupiter-engine</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>com.github.RMLio</groupId>
			<artifactId>rmlmapper-java</artifactId>
			<version>4.5.1</version>
		</dependency>
	</dependencies>

```

**Comment by bjdmeest** [comment]
I'm personally not very experienced in Spring, but I do know that we had to explicitly add the maven repository in our POM, also due to some conflicting dependencies, please see https://github.com/RMLio/rmlmapper-java/blob/df19dc309ca25ce4efa51f410ce51837cd2514f6/pom.xml#L18

Could you please first see whether this solves the issue?

**Comment by bjdmeest** [comment]
Also please see https://github.com/RMLio/rmlmapper-java/issues/32, this feels like a duplicate

**Comment by istoony** [comment]
I also saw #32 but there was not a solution.
Yesterday I finally solved the problem, here what I've done:

1. download the source code
2. packaging it by using "mvn package". This operation creates two different jar files, the first has around 60Mb and the second 150kb. 
3. Include the jar with 150Kb in my spring application:
`
<dependency>
       <groupId>rmlmapper</groupId>
	<artifactId>rmlmapper</artifactId>
	<version>4.5.1</version>
	<scope>system</scope>
	<systemPath>${project.basedir}/src/main/lib/original-rmlmapper-4.5.1-r140.jar</systemPath>
</dependency>
`
4. manually include the dependency that I need in my pom. For example when I try tu initialize
`QuadStore outputStore = new RDF4JStore();`
I need to include in pom
`<dependency>
	<groupId>org.eclipse.rdf4j</groupId>
	<artifactId>rdf4j-runtime</artifactId>
	<version>2.5.1</version>
</dependency>`

Thanks everyone. I hope this little tutorial will be helful for someone
